plugins/field-grid-dropdown/src/grid.ts

```diff
@@ -0,0 +1,282 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+import {GridItem} from './grid_item';
+
+/**
+ * Class for managing a group of items displayed in a grid.
+ */
+export class Grid {
+  /** Mapping from grid item ID to grid item. */
+  private itemIndices = new Map<string, number>();
+
+  /** List of items displayed in this grid. */
+  private items = new Array<GridItem>();
+
+  /** Root DOM element of this grid. */
+  private root: HTMLDivElement;
+
+  /** Identifier for keydown handler to be unregistered in dispose(). */
+  private keyDownHandler: Blockly.browserEvents.Data | null = null;
+
+  /** Identifier for pointermove handler to be unregistered in dispose(). */
+  private pointerMoveHandler: Blockly.browserEvents.Data | null = null;
+
+  /** Function to be called when an item in this grid is selected. */
+  private selectionCallback?: (selectedItem: GridItem) => void;
+
+  /**
+   * Creates a new Grid instance.
+   *
+   * @param container The parent element of this grid in the DOM.
+   * @param options A list of MenuOption objects representing the items to be
+   *     shown in this grid.
+   * @param columns The number of columns to display items in.
+   * @param rtl True if this grid is being shown in a right-to-left environment.
+   * @param selectionCallback Function to be called when an item in the grid is
+   *     selected.
+   */
+  constructor(
+    container: HTMLElement,
+    options: Blockly.MenuOption[],
+    private readonly columns: number,
+    private readonly rtl: boolean,
+    selectionCallback: (selectedItem: GridItem) => void,
+  ) {
+    this.selectionCallback = selectionCallback;
+
+    this.root = document.createElement('div');
+    this.root.className = 'blocklyGrid';
```

Comment:
If this is a new name, use `blocklyFieldGrid` to distinguish from the workspace background grid.

Reply:
Done.